### PR TITLE
Expose a `native_handle` property on `WindowBase`

### LINKED ISSUE #8597: Expose a method on `kivy.core.window` to get the window handle
Sometimes, and #8596 is one of these cases, we need to get the native Window handle.

With SDL2 (our main window core provider) that could be done by exposing the info retrieved via `SDL_GetWindowWMInfo`, but the API should be provider-agnostic and be ready for the next decade of Window providers.

### PR DESCRIPTION
**Pull request by DataTriny**
Maintainer merge checklist
* [x] Title is descriptive/clear for inclusion in release notes.
* [x] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [x] Added to the milestone version it was merged into.
* [x] **Unittests** are included in PR.
* [x] Properly documented, including `versionadded`, `versionchanged` as needed.

Closes #8597 

Retrieving a pointer to the `NSWindow` will also be required for AccessKit integration, but the native code is not present at the moment.

### CODE REVIEW

File: kivy/core/window/_window_sdl2.pyx
Comment (misl6) [inline kivy/core/window/_window_sdl2.pyx:513]: We're trying to avoid `IF/ELSE` and `DEF` on new code (and slowly migrating during changes required for `3.0.0`) as is deprecated.

See: http://docs.cython.org/en/latest/src/userguide/migrating_to_cy30.html#deprecation-of-def-if

**Review by misl6 — CHANGES_REQUESTED** [review]
Hi @DataTriny !

Thank you for taking care of it.

I've left a comment regarding a Cython deprecation, please consider taking care of it so we do not introduce deprecated code.

Regarding missing platforms: I will open a separate issue to target it on a separate PR, but will be quite straightforward as `SDL_GetWindowWMInfo` already provides the info we need.

What do you think about adding a test that exercises this new feature?

File: kivy/tests/test_window_base.py
Comment (DataTriny) [inline kivy/tests/test_window_base.py:144]: This check is different from the one in the method being tested, but this is what is used in `test_window_info.py` so I figured maybe your CI expects this exact configuration.

File: kivy/tests/test_window_base.py
Comment (DataTriny) [inline kivy/tests/test_window_base.py:132]: I guess another property of native handles is that they should not change over time, but I really don't know how to exercise that in a test.

Comment (misl6) [reply kivy/tests/test_window_base.py:132]: Checking if we're getting a value when we expect one is fine for me ATM.

**Review by misl6 — APPROVED** [review]
Let's wait for the CI jobs to complete the run, but LGTM.
Thank you!

**Pull request merged**
